Homework_2.playground/Contents.swift

```diff
@@ -0,0 +1,98 @@
+class Stack {
+    private let linkedList = LinkedList()
+    
+    func push(element: Int) {
+        linkedList.append(element)
+    }
+    
+    func pop() -> Int? {
+        return linkedList.removeLast()
+    }
+    
+    func top() -> Int? {
+        return linkedList.last
+    }
+}
+
+class LinkedList {
+    private var head: Frame?
+    var last: Int? {
+        return head?.data
+    }
+    
+    func append(_ element: Int) {
+        head = Frame(data: element, nextFrame: head)
```

Comment:
Let's imagine the following code:
```swift
private let linkedList = LinkedList()
linkedList.append(a)
linkedList.append(b)
```

You have the `nextFrame` property so I'd expect to see have the following linked list (after the execution the corresponding line of code):
```
head -> nil
head -> [a] -> nil
head -> [a] -> [b] -> nil
```
In your case the picture is:
```
head -> nil
head -> [a] -> nil
head -> [b] -> [a] -> nil
```
This can be confusing, please fix it.
Does that make sense?